syncd/FlexCounter.cpp

```diff
@@ -925,6 +905,45 @@ class CounterContext : public BaseCounterContext
         auto checkAndUpdateBulkCapability = [&](const std::vector<StatType> &counter_ids, const std::string &prefix, uint32_t bulk_chunk_size)
         {
             BulkContextType ctx;
+            // Check bulk capabilities again
+            std::vector<StatType> supportedBulkIds;
+            if (m_supportedBulkCounters.empty())
+            {
+                querySupportedCounters(rids[0], statsMode, m_supportedBulkCounters);
```

Comment:
why you are passing member here ?

Reply:
Now we have two set of capabilities 
- single call capabilities 
- bulk call capacities 

We leverage the function to fetch both set of capabilities 
We have to pass the member to distinguish them

Reply:
In the function bulkAddObject, a brief flow is the following 
1. Check single call capacities and remove the counters that don't support single call.
   In this step the function is called with m_supportedCounters as an argument.
2. Check bulk call capacities and use single call for the counters that don't support bulk call capacities (for sure they support single call otherwise they will be removed by step 1)
   In this step the function is called with m_supportedBulkCounters as an argument.

Reply:
ok, from first point it seems weird, that function that has access is getting member as parameter, usually that in case when function is static, anyway return status of querySupportedCounters is not checked

Reply:
the function `querySupportedCounters` applies the same logic on single or bulk capabilities and stores the supported counters in `m_supportedCounters` or `m_supportedBulkCounters` respectively.
we can either use a control flag to distinguish single/bulk or pass the member directly. Looks like it's more straightforward to pass the member directly. wdyt?
we can add return status check of the function.